### ISSUE TITLE
Add option to choose mocking library

### ISSUE THREAD

**Issue by landure** [paul-thebaud/phpunitgen-core#28]
### Subject of the issue

By default, PhpUnitGen uses Mockery for object mocking. Mockery doesn't support Psalm [Templating](https://psalm.dev/docs/annotating_code/templated_annotations/) (generic typing).

PHPUnit has mocking features supporting generic typing. It would be nice for PhpUnitGen to allow choosing the mocking library used.

### Your environment

* version of `phpunitgen/core`: 3.1.0
* version of `phpunitgen/console`: 5.0.0

### Steps to reproduce

Generate a test with PhpUnitGen where mocking is used.

### Expected behaviour

PhpUnitGen configuration file has a `mockingLibrary` setting.

- When set to `mockery`, it generates code using Mockery (current behaviour).
- When set to `phpunit`, it uses PHPUnit mocking feature, which supports psalm generic typing:

```php
        $expected = $this->getMockBuilder(MyCollection::class)->getMock();
```

### Actual behaviour

PhpUnitGen use Mockery for mocking:

```php
        $expected = Mockery::mock(MyCollection::class);
```

Since Mockery doesn't support Psalm generic typing, this trigger [Invalid Argument](https://psalm.dev/docs/running_psalm/issues/InvalidArgument/) warnings:

```php
 "Argument 1 of App\Document\\MyClass::setCollection expects Doctrine\\Common\\Collections\\Collection<array-key, object>, but Mockery\\LegacyMockInterface provided",
```

**Comment by paul-thebaud** [comment]
Hello @landure,

You can already change the mock generation library to use by defining an `implementations` key inside your configuration file:

```php
use PhpUnitGen\Core\Contracts\Generators\MockGenerator;
use PhpUnitGen\Core\Generators\Mocks\PhpUnitMockGenerator;
use PhpUnitGen\Core\Generators\Tests\DelegateTestGenerator;

return [
    'implementations' => array_merge(DelegateTestGenerator::implementations(), [
        MockGenerator::class => PhpUnitMockGenerator::class,
    ]),
];
```

Is this solving your problem?

**Comment by paul-thebaud** [comment]
@landure, fixed in [`phpunitgen-core@4.0.0`](https://github.com/paul-thebaud/phpunitgen-core/releases/tag/4.0.0) and [`phpunitgen-console@6.0.0`](https://github.com/paul-thebaud/phpunitgen-console/releases/tag/6.0.0).

**Comment by landure** [comment]
Thank you for your work.
